query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -149,8 +149,13 @@ private String convertFilterToString(
               handleValueConversionForLiteralExpression(filter.getLhs(), filter.getRhs());
           builder.append(operator);
           builder.append("(");
-          builder.append(
-              convertExpressionToString(filter.getLhs(), paramsBuilder, executionContext));
+          if (isMapColumnName(filter.getLhs())) {
+            // supports Like operator on map key column
+            builder.append(convertExpressionToMapKeyColumn(filter.getLhs()));
```

Comment:
On second thought, this does a regex search only on the field's keys and not on its values. Should we do a regex search on the field's values as well here, or should we introduce new operators like - `KEY_LIKE` and `VALUE_LIKE` and leave the `LIKE` operator unchanged?

Reply:
We already support operations on values via attribute expressions. We don't really want to introduce new operators, but the only potentially useful scenariosI could think of that we would be missing would be extensions of the `CONTAINS_KEY` operator - e.g. `NOT_CONTAINS_KEY` and `CONTAINS_KEY_LIKE`.

Reply:
@GurtejSohi `CONTAINS_KEY_LIKE` is what i think we need

Reply:
Removed these changes and added a new operator: `CONTAINS_KEY_LIKE `

Reply:
> We don't really want to introduce new operators, but the only potentially useful scenariosI could think of that we would be missing would be extensions of the CONTAINS_KEY operator - e.g. NOT_CONTAINS_KEY and CONTAINS_KEY_LIKE.

Actually, we already have `NOT_CONTAINS_KEY` operator.

Reply:
> Actually, we already have NOT_CONTAINS_KEY operator.

I believe we implemented at QS but haven't exposed it up to the platform API layer.